api/types/network/network.go

```diff
@@ -44,6 +44,7 @@ type EndpointSettings struct {
 	GlobalIPv6Address   string
 	GlobalIPv6PrefixLen int
 	MacAddress          string
+	DNS                 []string
```

Comment:
I dont think we should expose the endpoint DNS settings to the docker API just yet.
I think we should reuse the daemon / container `--dns` configurations as is and can set the endpoint level DNS settings at libnetwork.

When we are ready to expose such functionality at the UX (and possibly across platforms), then we can enable this at the API level. So, my request is to remove this for now and we can consider including it at a later point.